docs/wiki/framework/arsenal-framework.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: wiki
+title: ACE Arsenal Framework
+description: Explains how to interact with the ACE Arsenal API.
+group: framework
+parent: wiki
+mod: ace
+version:
+  major: 3
+  minor: 12
+  patch: 0
+---
+
+<div class="panel callout">
+    <h5>BI functions and variables from BI Virtual Arsenal DO NOT apply to ACE Arsenal!</h5>
+</div>
+
+<div class="panel callout">
+    <h5>Note:</h5>
+    <p>Units and objects that can have ACE Arsenal added to them will be called "Boxes" in this documentation.</p>
+</div>
+
+To quickly add a full ACE Arsenal to a box for all clients (do not use in init fields):  
+`[_box, true] call ace_arsenal_fnc_initBox`  
+`_box` being the object you wish to add ACE Arsenal to.
+
+## 1. Virtual items
+
+### 1.1 Adding virtual items
+
+`ace_arsenal_fnc_addVirtualItems`
+```cpp
+ * Arguments:
+ * 0: Box <OBJECT>
+ * 1: Items <ARRAY of strings> or <BOOL>
+ * 2: Add globally <BOOL>
+```
+
+Passing an array of strings (classnames) will add each one of those items to the specified box, passing true will add ALL items that are compatible with ACE Arsenal (that sorting is done on game startup).
+
+Examples:
+
+- `[_box, ["item1", "item2", "itemN"]] call ace_arsenal_fnc_addVirtualItems`
+- `[_box, true, false] call ace_arsenal_fnc_addVirtualItems`
+
+### 1.2 Removing virtual items
+
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Items <ARRAY of strings> <BOOL>
+ * 2: Add globally <BOOL>
+ ```
+
+Like adding virtual items, passing an array of string (classnames) will remove each ones of those items, however passing true will remove all virtual items and also remove the interaction to access ACE Arsenal.
+
+Examples:
+
+- `[_box, ["item1", "item2", "itemN"]] call ace_arsenal_fnc_removeVirtualItems`
+- `[_box, true, false] call ace_arsenal_fnc_removeVirtualItems`
+
+## 2. Access to ACE Arsenal
+
+### 2.1 Adding ACE Arsenal to a box
+
+`ace_arsenal_fnc_initBox`
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Items <BOOL> or <ARRAY>
+ * 2: Initialize globally <BOOL>
+```
+
+This will add the virtual items passed as arguments and add an ACE interaction to open ACE Arsenal.
+
+Examples:
+
+- `[_box, ["MyItem1", "MyItem2", "MyItemN"]] call ace_arsenal_fnc_initBox`
+- `[_box, true] call ace_arsenal_fnc_initBox`
+- `[_box, false, false] call ace_arsenal_fnc_initBox`
+
+Passing an empty array or `false` will still add an interaction but no additional virtual items will be added.  
+Please note that at least one virtual item need to be added otherwise ACE Arsenal will not open.
+
+if you wish to open a full ACE Arsenal on yourself or open ACE Arsenal via a custom action you can use `ace_arsenal_fnc_openBox`.
+
+`ace_arsenal_fnc_openBox`
+```cpp
+ * 0: Box <OBJECT>
+ * 1: Unit to open ACE Arsenal on <OBJECT>
+ * 2: Ignore virtual items and fill ACE Arsenal <BOOL>
+```
+
+Examples:
+
+- `[_box, player] call ace_arsenal_fnc_openBox`
+- `[player, player, true] call ace_arsenal_fnc_openBox`
+
+In the second example a full ACE Arsenal will be opened on the player.
```

Comment:
and in the first?

Reply:
No, in the first the "box" virtual items would be used, in the second example the box is the player and the item list is retrieved from the cache.

Reply:
i meant: people might want a description for the first example too